src/BowlingScorecard.js

```diff
@@ -0,0 +1,143 @@
+// const Frame = require('./Frame.js');
+
```

Comment:
Your BowlingScorecard class has a few responsibilities: -it increments the frame, it stores the frames, it checks rolls are valid, it applies bonuses etc. I think you could extract another class from this class.